toplevel/topcommon.ml

```diff
@@ -288,7 +288,7 @@ let load_topdirs_signature () =
   let compiler_libs =
     Filename.concat Config.standard_library "compiler-libs" in
   let topdirs_cmi = Filename.concat compiler_libs "topdirs.cmi" in
-  if Sys.file_exists topdirs_cmi then
+  if not !Clflags.no_std_include && Sys.file_exists topdirs_cmi then
     ignore (Env.read_signature "Topdirs" topdirs_cmi)
```

Comment:
I am increasingly convinced a better hackish fix is to catch the `Cmi_format`.Error _` exception here, and fails with a fatal error which should be more informative, relatively painless to fix whenever the error arises, and invisible for nearly all users.